File: template_byClass/${className}-ewandian-mybatis-postgresql/angular/services/${className}Service.js
<#include "/macro.include"/>
<#assign className = clazz.className>
<#assign classNameLower = className?uncap_first>
'use strict';
/**
 * @author：Tian
 * @date: ${now?string('yyyy/MM/dd')}
 */
define([], function () {
    var customersFactory = function ($http,$q) {
        var serviceBase = '/${namespace}Service/${classNameLower}/',
            factory = {};

        /** 获取${className}列表 */
        factory.list = function (vo) {
            var d = $q.defer();
            $http.post(serviceBase+'selectPageList',vo)
                .success(function (response) { d.resolve(response); })
                .error(function (response) {d.reject(response);});
            return d.promise;
        }

        /**
         * 保存${className}（包括新增，更新）
         * @param ${classNameLower}
         */
        factory.save = function (vo) {
            var d = $q.defer();
            $http.post(serviceBase+(vo['${clazz.fields[0].fieldName}'] ? 'updateOne' : 'insertOne'), vo)
                .success(function (response) { d.resolve(response); })
                .error(function (response) {d.reject(response);});
            return d.promise;
        };

        /**
         * 删除${className}，级联删除${className}所持Card
         * @param id 被删除的${classNameLower}的ID
         */
        factory.remove = function (id) {
            var d = $q.defer();
            $http.post(serviceBase+'deleteById/' + id)
                .success(function (response) { d.resolve(response); })
                .error(function (response) {d.reject(response);});
            return d.promise;
        };

        return factory;
    };

    angular.module('${basepackage}.${namespace}.${classNameLower}Service',[]).factory('${classNameLower}Service', customersFactory);
});

File: repository/itownet-template_byClass/byClass/service/${className}Service.java
<#include "/java_copyright.include">
<#assign className = clazz.className>
<#assign classNameLower = className?uncap_first>
package ${basepackage}.${namespace}.service;


import ${basepackage}.model.${namespace}.${className};

import java.util.List;

/**
 * 
 * @Description: 业务层接口

 * @author: ???
 */
public interface ${className}Service {

	/**
	 *
	 * @Title: selectAll
	 * @Description: 查找所有记录
	 */
	List<${className}> selectAll();

	/**
	 *
	 * @Title: save

	 * @Description: 保存
	 */
	void save(${className} ${classNameLower});

	/**
	 *
	 * @Title: update

	 * @Description: 更新
	 */
	int update(${className} ${classNameLower});

	/**
	 *
	 * @Title: selectByPK

	 * @Description: 查询
	 */
	${className} selectByPK(${clazz.fields?first.javaType} ${clazz.fields?first.fieldName});

	/**
	 *
	 * @Title: delete
	 *
	 * @Description: 删除
	 */
	int delete(${clazz.fields?first.javaType} ${clazz.fields?first.fieldName});

}
